src/systemctl/systemctl-show.c

```diff
@@ -419,12 +421,20 @@ static void print_status_info(
                     STRPTR_IN_SET(i->active_state, "activating")          ? i->inactive_exit_timestamp :
                                                                             i->active_exit_timestamp;
 
-        if (timestamp > 0 && timestamp < USEC_INFINITY)
+        if (timestamp > 0 && timestamp < USEC_INFINITY) {
                 printf(" since %s; %s\n",
                        FORMAT_TIMESTAMP_STYLE(timestamp, arg_timestamp_style),
                        FORMAT_TIMESTAMP_RELATIVE(timestamp));
-        else
+                if (STRPTR_IN_SET(i->active_state, "active") &&
+                        i->runtime_max_sec < USEC_INFINITY) {
+                        until_timestamp = timestamp + i->runtime_max_sec;
+                        printf("      Until: %s; %s\n",
+                                FORMAT_TIMESTAMP_STYLE(until_timestamp, arg_timestamp_style),
+                                FORMAT_TIMESTAMP_RELATIVE(until_timestamp));
+                        }
```

Comment:
indentation of the closing bracket here is off.. should be place under the `if`, since that's where the block was opened